query-engine/driver-adapters/executor/src/bench.ts

```diff
@@ -0,0 +1,204 @@
+    });
+    process.stdin.on("error", reject);
+  });
+
+  const url = process.env.DATABASE_URL;
+  if (url == null) {
+    throw new Error("DATABASE_URL is not defined");
+  }
+  const pg = await pgAdapter(url);
+  const withErrorCapturing = bindAdapter(pg);
+
+  const { recorder, replayer } = recording(withErrorCapturing);
+  await recordQueries(recorder, datamodel, prismaQueries);
+  await benchMarkQueries(replayer, datamodel, prismaQueries);
+}
+
+async function recordQueries(
+  adapter: DriverAdapter,
+  datamodel: string,
+  prismaQueries: any
+): Promise<void> {
+  const qe = await initQeWasmBaseLine(adapter, datamodel);
+  await qe.connect("");
+
+  for (const prismaQuery of prismaQueries) {
+    const { description, query } = prismaQuery;
+    const res = await qe.query(JSON.stringify(query), "", undefined);
+
+    const errors = JSON.parse(res).errors;
+    if (errors != null && errors.length > 0) {
+      throw new Error(
+        `Query failed for ${description}: ${JSON.stringify(res)}`
+      );
+    }
+  }
+
+  await qe.disconnect("");
+}
+
+async function benchMarkQueries(
+  adapter: DriverAdapter,
+  datamodel: string,
+  prismaQueries: any
+) {
+  const napi = await initQeNapiCurrent(adapter, datamodel);
+  await napi.connect("");
+  const wasmCurrent = await initQeWasmCurrent(adapter, datamodel);
+  await wasmCurrent.connect("");
+  const wasmBaseline = await initQeWasmBaseLine(adapter, datamodel);
+  await wasmBaseline.connect("");
+  const wasmLatest = await initQeWasmLatest(adapter, datamodel);
+  await wasmLatest.connect("");
+
+  for (const prismaQuery of prismaQueries) {
+    const { description, query } = prismaQuery;
+
+    const adapters = {
+      Napi: napi,
+      "WASM Current": wasmCurrent,
+      "WASM Baseline": wasmBaseline,
+      "WASM Latest": wasmLatest,
+    };
+
+    for (const [adapterName, adapter] of Object.entries(adapters)) {
+      const res = await adapter.query(JSON.stringify(query), "", undefined);
+      const errors = JSON.parse(res).errors;
+      if (errors != null && errors.length > 0) {
+        throw new Error(
+          `${adapterName} - Query failed for ${description}: ${JSON.stringify(
+            res
+          )}`
+        );
+      }
+    }
+  }
+
+  try {
+    for (const prismaQuery of prismaQueries) {
+      const { description, query } = prismaQuery;
+      const jsonQuery = JSON.stringify(query);
+
+      group(description, () => {
+        bench(
+          "Web Assembly: Baseline",
+          async () => await wasmBaseline.query(jsonQuery, "", undefined)
+        );
+        bench(
+          "Web Assembly: Latest",
+          async () => await wasmLatest.query(jsonQuery, "", undefined)
+        );
+        baseline(
+          "Web Assembly: Current",
+          async () => await wasmCurrent.query(jsonQuery, "", undefined)
+        );
+        bench(
+          "Node API: Current",
+          async () => await napi.query(jsonQuery, "", undefined)
+        );
```

Comment:
We could probably extact that recurrent `""` input argument into a top-level string variable `const trace = '';`, to remind future code readers what that argument is for.

Reply:
(This suggestion applies to other parts of this file as well)